pkg/tasks/c1api/full_sync.go

```diff
@@ -140,3 +146,35 @@ func newFullSyncTaskHandler(task *v1.Task, helpers fullSyncHelpers, skipFullSync
 		skipFullSync: skipFullSync,
 	}
 }
+
+func uploadDebugLogs(ctx context.Context, helper fullSyncHelpers) error {
+	l := ctxzap.Extract(ctx)
+
+	debugfilelocation := filepath.Join(helper.TempDir(), "debug.log")
+
+	_, err := os.Stat(debugfilelocation)
+	if err == nil {
+		debugfile, err := os.Open(debugfilelocation)
+		if err != nil {
+			return err
+		}
+		defer debugfile.Close()
+
+		l.Info("uploading debug logs", zap.String("file", debugfilelocation))
+		err = helper.Upload(ctx, debugfile)
+
+		if err != nil {
+			return err
+		}
+		defer func() {
+			err := os.Remove(debugfilelocation)
+			if err != nil {
+				l.Error("failed to delete file with debug logs", zap.Error(err), zap.String("file", debugfilelocation))
+			}
+		}()
+	} else if errors.Is(err, os.ErrNotExist) {
+		l.Warn("debug log file does not exists", zap.Error(err))
+	}
```

Comment:
The way this is written it seems like if `os.Stat()` errors for some reason, we just return nil.
I think it might make sense to do something like.
```go
_, err := os.Stat(debugfilelocation)
if err != nil {
  if errors.Is(err, os.ErrNotExist) {
    l.Warn("debug log file does not exists", zap.Error(err))
    return nil
  }
  return err
}
```
I think if the intention is to fail silently, it would be helpful to leave a comment.

Reply:
> The way this is written it seems like if os.Stat() errors for some reason, we just return nil.

I just thought that we should not care about any other error than `os.ErrNotExist`, and given how the top level is written, maybe returning the error from `os.Stat` won't break anything (who knows 🤷🏻‍♂️)